app/controllers/stormpath/rails/login/create_controller.rb

```diff
@@ -44,7 +48,11 @@ def set_cookies
         end
 
         def serialized_account
-          AccountSerializer.to_h(form.authentication_result.account)
+          if social_login?
+            AccountSerializer.to_h(form.account)
+          else
+            AccountSerializer.to_h(form.authentication_result.account)
+          end
```

Comment:
can we make this as one interface? So both return account maybe? Can you check if we need an authentication result?